python/lsst/ts/cRIOpy/GUI/CustomLabels.py

```diff
@@ -307,6 +333,96 @@ def setValue(self, value):
             self.setText("<font color='green'>OK</font>")
 
 
+class WarningButton(QPushButton):
+    """Displays WARNING/OK button. When clicked, displays window with values in
+    the signal. Button is color-coded (red for problems, green for OK values).
+
+    Parameters
+    ----------
+    comm : `SALComm`
+        SALComm object representing the signals
+    topic : `str`
+        Topic name. Should be event/telemetry name without leading evt_ (for events).
+    field : `str`, optional
+        Field from topic used to display button state. Defaults to anyWarning
+    """
+
+    def __init__(self, comm, topic, field="anyWarning"):
+        super().__init__("---")
+        self.comm = comm
+        self._topic = topic
+        self._field = field
+        getattr(comm, topic).connect(self._data)
+        self.window = None
+        self.clicked.connect(self._showWindow)
+
+    @Slot(map)
+    def _data(self, data):
+        self.setValue(getattr(data, self._field))
+
+    def _showWindow(self):
+        if self.window is None:
+            self.window = EventWindow(self.comm, self._topic)
+        self.window.show()
+
+    def setValue(self, value):
+        """Sets formatted value. Color codes WARNING (red)/OK (green).
+
+        Parameters
+        ----------
+        value : `bool`
+            Current (=to be displayed) variable value. True means warning.
+        """
+        pal = self.palette()
+        if value:
+            self.setText("WARNING")
+            pal.setColor(QPalette.Button, Qt.red)
+        else:
+            self.setText("OK")
+            pal.setColor(QPalette.Button, Qt.green)
+        self.setPalette(pal)
+
+
+class InterlockOffLabel(QLabel):
+    """Displays PROBLEM/OK. Constructor can be passed parameters allowing
+    connection to a Signal emitted when warning value changes.
+
+    Parameters
+    ----------
+    signal : `Signal`, optional
+        When not None, given signal will be connected to method calling
+        setValue with a field from signal data. Field is the second argument.
+        Defaults to None.
+    field : `str`, optional
+        When specified (and signal parameter is provided), will use this field
+        as fieldname from data arriving with the signal. Defaults to
+        "anyWarning".
+    """
+
+    def __init__(self, signal=None, field="anyWarning"):
+        super().__init__("---")
+        if signal is not None:
+            self._field = field
+            signal.connect(self._data)
+
+    @Slot(map)
+    def _data(self, data):
+        self.setValue(getattr(data, self._field))
+
+    def setValue(self, value):
+        """Sets formatted value. Color codes WARNING (red)/OK (green).
+
+        Parameters
+        ----------
+        value : `bool`
+            Current (=to be displayed) variable value. False is OK, True is a problem.
```

Comment:
I think you could have a better name than `value`, which is a bool here. I thought the use of `value` word will be more wide than bool only.

Reply:
Renamed to interlockOff